code/script.js

```diff
@@ -220,106 +226,154 @@ const generateBoard = () => {
       </div>
     `
   })
-}
+};
 
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
+};
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
-}
+
+  generateBoard() //Load all the characters
+  setSecret() //Generates a new secret person to guess on at the start
+};
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  const category = questions.options[questions.selectedIndex].parentNode.label 
+  const selectedOption = questions.options[questions.selectedIndex];
+  // This variable stores the actual value of the question we've selected.
+  const value = selectedOption.value; 
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
-}
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  const {category, value} = currentQuestion
+  let keep
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
+  // Compares the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).Decides who to keep or remove  based on that
   if (category === 'hair' || category === 'eyes') {
-
+    keep = secret[category] === value;
+    // Defines the condition for 'accessories' or 'other' category.  
   } else if (category === 'accessories' || category === 'other') {
-
+      keep = secret[category].includes(value);
   }
-}
+  // This invoke filterCharacters
+  filterCharacters(keep)
+};
 
-// It'll filter the characters array and redraw the game board.
+// This will filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
+  const {category, value} = currentQuestion
+  
   // Show the correct alert message for different categories
   if (category === 'accessories') {
```

Comment:
You mentioned in your comment to the pull request that this if/else statement could have been made shorter, do you have any thoughts on what this could look like?